src/js/client/editprefset-productList-widget.js

```diff
@@ -0,0 +1,149 @@
+ * the left hand side for navigation and adding new products to an NP Set.
+ */
+/* global lunr */
+"use strict";
+
+var gpii  = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.devpmt");
+fluid.registerNamespace("gpii.devpmt.productList");
+
+/**
+ *  Main infusion component for a filterable product list.
+ */
+fluid.defaults("gpii.devpmt.productListWidget", {
+    gradeNames: ["gpii.handlebars.templateAware", "gpii.binder.bindOnCreate", "gpii.binder.bindOnDomChange"],
+    model: {
+        productListFilter: "",
+        allSolutionsSorted: []
+    },
+    modelListeners: {
+        productListFilter: {
+            func: "{that}.filterProductList"
+        },
+        allSolutionsSorted: {
+            func: "{that}.updateLunrIndex"
+        }
+    },
+    selectors: {
+        productListFilterContainer: "#filter-container",
+        productListItems: ".pmt-sidebar-toc-product",
+        productListLinks: ".pmt-sidebar-product-link"
+    },
+    bindings: {
+    },
+    templates: {
+        initial: "editprefset-productList-widget"
+    },
+    components: {
+        filter: {
+            type: "gpii.devpmt.filterWidget",
+            createOnEvent: "onMarkupRendered",
+            container: "{that}.dom.productListFilterContainer",
+            options: {
+                selectors: {
+                    initial: "#common-filter-area"
+                },
+                model: {
+                    filterText: "{productListWidget}.model.productListFilter"
+                }
+            }
+        }
+    },
+    lunrIndex: null,
+    invokers: {
+        renderInitialMarkup: {
+            func: "{that}.renderMarkup",
+            args: ["initial", "{that}.options.templates.initial", "{that}.model"]
+        },
+        updateLunrIndex: {
+            funcName: "gpii.devpmt.productList.updateProductListLunrIndex",
+            args: ["{that}"]
+        },
+        filterProductList: {
+            funcName: "gpii.devpmt.productList.filterProductList",
+            args: ["{that}", "{that}.dom.productListItems", "{that}.model.productListFilter"]
+        },
+        // Used when clicking on the product, or otherwise activating it from the list.
+        selectProduct: {
+            funcName: "gpii.devpmt.productList.selectProduct",
+            args: ["{that}", "{gpii.devpmt.editPrefs}", "{arguments}.0"]
+        }
+    },
+    events: {
+        activateProduct: null
+    },
+    listeners: {
+        "onMarkupRendered": [
+            {
+                "this": "{that}.dom.productListLinks",
+                "method": "click",
+                args: ["{that}.selectProduct"]
+            }
+        ]
+    }
+});
+
+gpii.devpmt.productList.selectProduct = function (that, devpmt, event) {
+    var appid = event.currentTarget.dataset.appid;
+    that.events.activateProduct.fire(appid);
+    // TODO Fix up the create order or figure out how to add a delayed listener,
```

Comment:
Totally unclear comment which I bet you can't interpret any more.

Reply:
bump.

Reply:
I had wanted the top level view component of the page to set up a listener to certain clicks in this subcomponent. But the creation of the subcomponent didn't occur until later after the page load, and I was having issues getting a listener setup after the fact.  But I think it's working fine as it is, and will remove the comment.